service/main.go

```diff
@@ -99,3 +84,28 @@ func setLogLevel(logger *log.Logger, logLevel string) {
 		*logger = level.NewFilter(*logger, level.AllowInfo())
 	}
 }
+
+func gitClient(env env.Vars, logger log.Logger) git.BasicClient {
+	var cl git.BasicClient
+	var err error
+
+	var opts []git.Option
+	if env.LogLevel == "DEBUG" {
+		opts = append(opts, git.WithProgressWriter(os.Stdout))
+	}
+
+	if env.GitAuthMethod == "https" {
```

Comment:
Cringed when I brought over the if / else, but the refactor to remove that seemed like unnecessary bloat on the Progress Writer change.